### PR TITLE
SG-29997 - FIRST PHASE Python2 removing

### PR DESCRIPTION
**Pull request by eduardoChaucaGallegos**
- ensure_* functions replaced by function from sgutils.py.
- six types replaced by native

### CODE REVIEW

File: shotgun_api3/lib/sgutils.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    For Python 3:
+      - `str` -> `str`
+      - `bytes` -> decoded to `str`
+    """
+    if not isinstance(s, (str, bytes)):
+        raise TypeError(f"not expecting type '{type(s)}'")
+    if isinstance(s, bytes):
+        s = s.decode(encoding, errors)
+    return s
```
Comment (julien-lang) [inline shotgun_api3/lib/sgutils.py:59]: Minor: the two functions don't use the same condition logic.

```
    if isinstance(s, str):
        return s

    elif isinstance(s, bytes):
        return s.decode(encoding, errors)
 
    raise TypeError(f"not expecting type '{type(s)}'")
```

**Review by julien-lang — COMMENTED** [review]
I see that you are adding the "ensure_" functions to a module but I don't see them being removed from the six module. Should not we do it?

**Review by carlos-villavicencio-adsk — APPROVED** [review]
LGTM

**Review by carlos-villavicencio-adsk — APPROVED** [review]

**Review by julien-lang — APPROVED** [review]

**Comment by mathbou** [comment]
Just a little reminder here, even if it's already merged; most of the work has [already been done](https://github.com/shotgunsoftware/python-api/pull/303) months ago

**Comment by eduardoChaucaGallegos** [comment]
> Just a little reminder here, even if it's already merged; most of the work has [already been done](https://github.com/shotgunsoftware/python-api/pull/303) months ago

@mathbou Thanks for your collaboration, we have considered your changes for this PR. We did a new PR because we are trying to take less risk and doing this python2 removal in phases.

**Pull request merged**